rust/src/lib.rs

```diff
@@ -327,248 +312,233 @@ pub extern "C" fn zcashlc_init_accounts_table_with_keys(
 /// - The memory referenced by `seed` must not be mutated for the duration of the function call.
 /// - The total size `seed_len` must be no larger than `isize::MAX`. See the safety documentation
 ///   of pointer::offset.
-/// - Call `zcashlc_free_keys` to free the memory associated with the returned pointer when
+/// - Call `zcashlc_free_binary_key` to free the memory associated with the returned pointer when
 ///   you are finished using it.
 #[no_mangle]
-pub unsafe extern "C" fn zcashlc_derive_extended_spending_keys(
+pub unsafe extern "C" fn zcashlc_derive_unified_spending_key(
     seed: *const u8,
     seed_len: usize,
-    accounts: i32,
+    account: i32,
     network_id: u32,
-) -> *mut FFIEncodedKeys {
+) -> *mut FFIBinaryKey {
     let res = catch_panic(|| {
         let network = parse_network(network_id)?;
         let seed = slice::from_raw_parts(seed, seed_len);
-        let accounts = if accounts > 0 {
-            accounts as u32
+        let account = if account >= 0 {
+            account as u32
         } else {
-            return Err(format_err!("accounts argument must be greater than zero"));
+            return Err(format_err!("account ID argument must be nonnegative"));
         };
 
-        Ok(FFIEncodedKeys::ptr_from_vec(
-            (0..accounts)
-                .map(|account| {
-                    let account = AccountId::from(account);
-                    UnifiedSpendingKey::from_seed(&network, seed, account)
-                        .map_err(|e| {
-                            format_err!("error generating unified spending key from seed: {:?}", e)
-                        })
-                        .map(move |usk| {
-                            let encoded = encode_extended_spending_key(
-                                network.hrp_sapling_extended_spending_key(),
-                                usk.sapling(),
-                            );
-                            FFIEncodedKey::new(account, encoded)
-                        })
-                })
-                .collect::<Result<_, _>>()?,
-        ))
+        let account = AccountId::from(account);
+        UnifiedSpendingKey::from_seed(&network, seed, account)
+            .map_err(|e| format_err!("error generating unified spending key from seed: {:?}", e))
+            .map(move |usk| {
+                let encoded = usk.to_bytes(Era::Orchard);
```

Comment:
Does this have orchard spending keys? It was mention that it shouldn't be the case in the arborist call of Sept 9th

https://github.com/zcash/librustzcash/issues/630

Reply:
This should be treated as an opaque binary value, so it doesn't matter at this point whether it contains Orchard keys or not. Since we no longer provide address derivation after this change (we now rely upon the wallet backend to derive addresses) we can be confident that the wallet will not provide an address where it can't successfully receive funds.